ReText/config.py

```diff
@@ -49,6 +49,26 @@ def updateButtonText(self):
 		else:
 			self.setText(self.defaultText)
 
+class DirectorySelectButton(QPushButton):
+	def __init__(self, parent, fileName):
+		QPushButton.__init__(self, parent)
+		self.fileName = fileName
+		self.defaultText = self.tr('(none)')
+		self.updateButtonText()
+		self.clicked.connect(self.processClick)
+
+	def processClick(self):
+		startDir = (QFileInfo(self.fileName).absolutePath()
+		            if self.fileName else '')
+		self.fileName = QFileDialog.getExistingDirectory(
+			self, self.tr('Select directory to open'), startDir)
+		self.updateButtonText()
+
+	def updateButtonText(self):
+		if self.fileName:
+			self.setText(QFileInfo(self.fileName).fileName())
+		else:
+			self.setText(self.defaultText)
```

Comment:
The `__init__` and `updateButtonText` methods of this class are copied from `FileSelectButton` without any changes. Maybe you can split them out into a base class, to avoid duplication?